### PR TITLE
feat: ✨ Approve/decline a single agreement round 2

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

- better layout for Review Cards
- move logix to react hooks module
- fixes render error on agreement meta data on review page and updates e2e tests
- adds batch api call for approving/declining change requests

Does not include

- confirmation modals or alerts

## Issue

#2349 

## How to test

1. goto agreement 9
2. make a budget change and send for approval
3. make a status change for the other budget line
4. review change requests and click link to approve page
5. should filter by action type and approve/decline the corresponding budget lines

## Screenshots

_If relevant, e.g. for a front-end feature_

## Definition of Done Checklist
- [X] OESA: Code refactored for clarity
- [X] OESA: Dependency rules followed
- [X] Automated unit tests updated and passed
- [X] Automated integration tests updated and passed
- [X] Automated quality tests updated and passed
- [X] Automated load tests updated and passed
- [X] Automated a11y tests updated and passed
- [X] Automated security tests updated and passed
- [X] 90%+ Code coverage achieved
- [-] Form validations updated

### CODE REVIEW

File: backend/openapi.yml [hunk 1]
```diff
@@ -1654,22 +1654,25 @@ paths:
                   type: integer
                 action:
                   type: string
-                  description: 'APPROVE or REJECT'
+                  description: "APPROVE or REJECT"
+                requestor_notes:
```
Comment (fpigeonjr) [inline backend/openapi.yml:1658]: added this and prettier went crazy 💅

File: frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    const processChangeRequests = (changeRequests, action) => {
+        let promises = changeRequests.map((changeRequest) => {
+            return reviewCR({
+                change_request_id: changeRequest.id,
+                action,
+                reviewer_notes: notes
+            })
+                .unwrap()
+                .then((fulfilled) => {
+                    console.log(`BLI: ${action}`, fulfilled);
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js]: fixed

File: frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    const processChangeRequests = (changeRequests, action) => {
+        let promises = changeRequests.map((changeRequest) => {
+            return reviewCR({
+                change_request_id: changeRequest.id,
+                action,
+                reviewer_notes: notes
+            })
+                .unwrap()
+                .then((fulfilled) => {
+                    console.log(`Change Request: ${action}`, fulfilled);
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js]: fixed

File: frontend/cypress/e2e/reviewChangeRequestsAtAgreementLevel.cy.js [hunk 4]
```diff
@@ -249,19 +301,20 @@ describe("Review Change Requests at the Agreement Level", () => {
...
-                            url: `http://localhost:8080/api/v1/agreements/${agreementId}`,
-                            headers: {
-                                Authorization: bearer_token,
-                                Accept: "application/json"
-                            }
-                        }).then((response) => {
-                            expect(response.status).to.eq(200);
-                        });
                     });
+                // TODO: unable to delete agreement?
```
Comment (fpigeonjr) [inline frontend/cypress/e2e/reviewChangeRequestsAtAgreementLevel.cy.js:305]: 😞 still a problema

File: frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+     * Handles the approval of a change request
+     * @param {'APPROVE' | 'REJECT'} action - The action to take (APPROVE or REJECT)
+     */
+    const handleApproveChangeRequests = async (action) => {
+        /**
+         * @type {ChangeRequest[]}
+         */
+        let changeRequests = [];
+        const BUDGET_APPROVE =
+            action === CHANGE_REQUEST_ACTION.APPROVE && changeRequestType === CHANGE_REQUEST_SLUG_TYPES.BUDGET;
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js:180]: will need these in the next round for modals and alerts

File: frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+            }
+        });
+    };
+    /**
+     * Handles the approval of a change request
+     * @param {Object[]} changeRequests - The change requests to approve
+     * @param {{APPROVE: string, DECLINE: string, CANCEL: string}} action - The action to take
+     */
+    const processChangeRequests = (changeRequests, action) => {
+        let promises = changeRequests.map((changeRequest) => {
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js:135]: 🔍 crux of the PR here

**Review by stevtek — APPROVED** [review]

**Pull request merged**